### ISSUE TITLE
Validate that either Version Name or Version Code is provided

### ISSUE THREAD

**Issue by viktorbenei** [bitrise-community/steps-change-android-versioncode-and-versionname#6]
If neither is provided then the step is pretty much just a "no-op" and can be confusing why it's Green even though it does not do anything.

Additionally, it's possible to just set the Offset input, expecting that it will offset the value set in the build.gradle file, but of course that offset is a different offset, that's not how the step works (the offset it only applied on the value provided as the step's input, the build.gradle file's original version infos are not used at all).

E.g.

```
+------------------------------------------------------------------------------+
| (2) change-android-versioncode-and-versionname@1.0.0                         |
+------------------------------------------------------------------------------+
| id: change-android-versioncode-and-versionname                               |
| version: 1.0.0                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: go                                                                  |
| time: 2018-02-07T15:35:14Z                                                   |
+------------------------------------------------------------------------------+
|                                                                              |

[34;1mConfigs:[0m
- BuildGradlePth: REDACTED/build.gradle
- NewVersionName: 
- NewVersionCode: 
- VersionCodeOffset: 100000000

[34;1mUpdating versionName and versionCode in: REDACTED/build.gradle[0m

[32;1m0 versionCode updated[0m
[32;1m0 versionName updated[0m
|                                                                              |
+---+---------------------------------------------------------------+----------+
| [32;1m✓[0m | [32;1mchange-android-versioncode-and-versionname@1.0.0[0m              | 2.16 sec |
+---+---------------------------------------------------------------+----------+
```

**Comment by KunamiPT** [comment]
What is the point of using a VersionCodeOffset if doens't use the gradle version code value?
If my current version code is 57 and I set:
- NewVersionCode: 58
- VersionCodeOffset: 1

What should I expect? I just want to increment +1 on my current version code automatically.

**Comment by KunamiPT** [comment]
AutoIncrement Shouldn't be sommething like this?

(...)
```
regexp.MustCompile(`^versionCode (?P<version_code>.*)`): func(line string, lineNum int, match []string) string {
			oldVersionCode := match[1]
			//finalVersionCode = oldVersionCode
			updatedLine := ""

			if configs.VersionCodeOffset != "" { //if configs.NewVersionCode != "" {
				//finalVersionCode = strconv.Itoa(newVersionCode)
                                //updatedLine = strings.Replace(line, oldVersionCode, finalVersionCode, -1)
				updatedLine = strings.Replace(line, oldVersionCode, oldVersionCode + offset , -1)
				updatedVersionCodes++
				log.Printf("updating line (%d): %s -> %s", lineNum, line, updatedLine)
			}

			return updatedLine
		},
```
(...)

**Comment by viktorbenei** [comment]
@KunamiPT this step is not designed to read/evaluate the build number, nor to bump it. It's meant to set it to a specific value.

Usual use case is to set it to `$BITRISE_BUILD_NUMBER`, which is the build number on bitrise.io
More infos why: https://stackoverflow.com/questions/33821137/build-versioning-in-continuous-delivery/33821876#33821876

**Comment by viktorbenei** [comment]
The offset is mainly used to specify an offset to this base build number, for simplicity (e.g. something like: `$BITRISE_BUILD_NUMBER` +100)

**Comment by BirmacherAkos** [comment]
We have released a new version [(1.1.0)](https://github.com/bitrise-community/steps-change-android-versioncode-and-versionname/releases/tag/1.1.0) of Change Android versionCode and versionName

Changes:
- If neither NewVersionCode or NewVersionName are provided the step will fail

**Comment by viktorbenei** [comment]
Thanks @BirmacherAkos , closing it! ;)
